src/main/java/seedu/duke/parsers/ModHappyParser.java

```diff
@@ -73,13 +83,27 @@ private Parser getCommandParser(String commandWord) throws UnknownCommandExcepti
         case (LIST_COMMAND_WORD):
             return new ListParser();
         case (ADD_COMMAND_WORD):
-            return new AddParser();
+            switch (parsedCommand.get(ARGUMENT).split(SPACE)[0]) {
+            case TASK:
+                return new AddTaskParser();
+            case MODULE:
+                return new AddModuleParser();
+            default:
+                throw new UnknownCommandException();
+            }
```

Comment:
I suggests to put the logic to AddParser, and let AddParser to decide whether it's module or task to keep ModHappyParser in consistent logical level. (May let AddParser decide whether return AddTaskParser or AddModuleParser to ModHappyParser)

Reply:
Shifted the logic to abstract class AddParser